### PR TITLE
feat(Config): Subreddit removal reason settings

### LINKED ISSUE #219: Removal Reason settings per subreddit
There's a few things we currently have user-side that could also exist in subreddit settings.

- [ ] Sticky removal message
- [ ] Lock removal message
- [ ] Remove as comment/PM/Modmail
- [x] Enable/Disable removal reason for posts (#224)
- [x] Enable/Disable removal reason for comments (#224)

### PR DESCRIPTION
**Pull request by SpyTec**
Ready for review 
Closes #219 

image

Clicking show advanced will display log-option
image

When removal option is forced
image

### CODE REVIEW

**Comment by SpyTec** [comment]
Do we want to add describing header as well?

**Comment by creesch** [comment]
> 
> 
> Do we want to add describing header as well?

Not sure what you mean with that?

**Comment by SpyTec** [comment]
> Not sure what you mean with that?
Such as the `Header` and `Footer`. Adding something like `Removal settings`

File: extension/data/modules/config.js [hunk 1]
```diff
@@ -94,56 +94,83 @@ function tbconfig () {
...
-                    </tr><tr>
-                    <td><textarea placeholder="Footer" class="tb-input edit-footer" >${TBHelpers.htmlEncode(unescape(configData.removalReasons.footer ? configData.removalReasons.footer : ''))}</textarea></td>
-                    </tr>
-                    <tr class="advanced-enable" ${(TBCore.advancedMode ? '' : 'style="display:none;"')}>
-                    <td><a href="javascript:;" class="show-advanced tb-general-button">show advanced settings</a></td>
-                    </tr>
-                    <tr class="rr-advanced">
-                    <td>
+                <div id="tb-removal-reason-settings">
+                    <h2></h2>
```
Comment (creesch) [inline extension/data/modules/config.js]:
```suggestion

```

File: extension/data/modules/config.js [hunk 2]
```diff
@@ -94,56 +94,83 @@ function tbconfig () {
...
+                    <span>Header</span>
```
Comment (creesch) [inline extension/data/modules/config.js]:
```suggestion
                    <h3>Removal reason header text</h3>
```

File: extension/data/modules/config.js [hunk 3]
```diff
@@ -94,56 +94,83 @@ function tbconfig () {
...
+                    <textarea placeholder="Header" class="tb-input edit-header" style="display:block;">${TBHelpers.htmlEncode(unescape(configData.removalReasons.header ? configData.removalReasons.header : ''))}</textarea>
+                    <span>Footer</span>
```
Comment (creesch) [inline extension/data/modules/config.js]:
```suggestion
                    <h3>Removal reason footer text</h3>
```

File: extension/data/modules/config.js [hunk 4]
```diff
@@ -94,56 +94,83 @@ function tbconfig () {
...
+                    <textarea placeholder="Footer" class="tb-input edit-footer" style="display:block;">${TBHelpers.htmlEncode(unescape(configData.removalReasons.footer ? configData.removalReasons.footer : ''))}</textarea>
+                    <select name="removal-option" id="removal-option">
```
Comment (creesch) [inline extension/data/modules/config.js]:
```suggestion
                    <h3>Removal reason actions<h3>
                    <select name="removal-option" id="removal-option">
```

File: extension/data/modules/config.js [hunk 5]
```diff
@@ -94,56 +94,83 @@ function tbconfig () {
...
+                                <label for="type-reply-none">Send nothing and log the removal.</label>
+                            </li>
+                            <li>
+                                <input class="action-lock-thread" type="checkbox" id="type-action-lock-thread" ${configData.removalReasons.typeLockThread ? 'checked' : ''}/>
+                                <label for="type-action-lock-thread">Lock the removed thread.</label>
+                            </li>
+                        </ul>
+                    </div>
+                    <a href="javascript:;" class="advanced-enable show-advanced tb-general-button" ${(TBCore.advancedMode ? '' : 'style="display:none;"')}>show advanced settings</a>
+                    <div class="rr-advanced">
```
Comment (creesch) [inline extension/data/modules/config.js]:
```suggestion
                    <h3 class="rr-advanced">Advanced settings</h3>
                    <div class="rr-advanced">
```

**Pull request merged**